x-pack/plugins/ml/public/application/components/anomalies_table/anomalies_table.js

```diff
@@ -148,11 +148,12 @@ export class AnomaliesTableInternal extends Component {
   };
 
   onTableChange = ({ page, sort }) => {
+    const pageCopy = { ...page };
     const { tableState, updateTableState } = this.props;
     const result = {
-      pageIndex: page && page.index !== undefined ? page.index : tableState.pageIndex,
-      pageSize: page && page.size !== undefined ? page.size : tableState.pageSize,
-      sortField: sort && sort.field !== undefined ? sort.field : tableState.sortField,
+      pageIndex: page && page.index !== undefined ? pageCopy.index : tableState.pageIndex,
+      pageSize: page && page.size !== undefined ? pageCopy.size : tableState.pageSize,
+      sortField: sort && sort.field !== undefined ? pageCopy.field : tableState.sortField,
```

Comment:
The actual problem is with `sortFIeld` and changes in your PR replace it with `pageCopy.field` which is always `undefined` because the updates come from the `sort` property. Check https://github.com/elastic/kibana/pull/98270 for more details.

Also important to note, in JavaScript, numbers and strings are no a reference type, hence creating`pageCopy` doesn't affect anything, because we access `index` and `size` by value at all times.